### PR TITLE
Makes chem dispenders upgradable once again!

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
[Changelogs]
Adds more emag chems to all dispensers!
Adds upgrade able tears to the chem dispenser
[why]
Make upgrading chemisty do something other then micro/recharge speed
More emag fun
Catwyn is a cat

### CODE REVIEW

**Comment by MimicFaux** [comment]
I feel like science upgrades and adding emag functionality should be split between two different PRs

**Comment by Trilbyspaceclone** [comment]
@MimicFaux 
Dosnt upgrade into emag options with max parts, still needs to be emaged first, sorry for the mix up!

File: code/modules/reagents/chemistry/machinery/chem_dispenser.dm [hunk 1]
```diff
@@ -497,13 +537,20 @@
...
+	upgrade_reagents3 = null
 	emagged_reagents = list(
 		"ethanol",
 		"iron",
+		"clownstears",
 		"minttoxin",
 		"atomicbomb",
-		"fernet"
+		"fernet",
+		"slimejelly",
```
Comment (silicons) [inline code/modules/reagents/chemistry/machinery/chem_dispenser.dm]: no, it's slime jelly for a reason, and just being able to machine out this is ridiculous when it does such high damage.

File: code/modules/reagents/chemistry/machinery/chem_dispenser.dm [hunk 2]
```diff
@@ -49,12 +49,30 @@
...
+	)
+
+	var/list/upgrade_reagents2 = list(
+		"acetone",
+		"phenol",
+		"diethylamine"
+	)
+
+	var/list/upgrade_reagents3 = list(
+		"omnizine",
```
Comment (silicons) [inline code/modules/reagents/chemistry/machinery/chem_dispenser.dm]: no, just no. too powerful.

File: code/modules/reagents/chemistry/machinery/chem_dispenser.dm [hunk 3]
```diff
@@ -49,12 +49,30 @@
...
+		"glycerol"
```
Comment (silicons) [inline code/modules/reagents/chemistry/machinery/chem_dispenser.dm]: no thank you infinite nitro bombs

File: code/modules/reagents/chemistry/machinery/chem_dispenser.dm [hunk 4]
```diff
@@ -49,12 +49,30 @@
...
+	)
+
 	var/list/emagged_reagents = list(
 		"space_drugs",
-		"morphine",
+		"sulfonal",
+		"frostoil",
 		"carpotoxin",
-		"mine_salve",
```
Comment (silicons) [inline code/modules/reagents/chemistry/machinery/chem_dispenser.dm:56]: why are you removing these

File: code/modules/reagents/chemistry/machinery/chem_dispenser.dm [hunk 5]
```diff
@@ -49,12 +49,30 @@
...
-		"toxin"
```
Comment (silicons) [inline code/modules/reagents/chemistry/machinery/chem_dispenser.dm:57]: this is actually useful, add it to the regular chem dispenser if you want but definitely do not remove

File: code/modules/reagents/chemistry/machinery/chem_dispenser.dm [hunk 6]
```diff
@@ -49,12 +49,30 @@
...
+
+	var/list/upgrade_reagents3 = list(
+		"omnizine",
+		"glycerol"
+	)
+
 	var/list/emagged_reagents = list(
 		"space_drugs",
-		"morphine",
+		"sulfonal",
```
Comment (silicons) [inline code/modules/reagents/chemistry/machinery/chem_dispenser.dm]: no, they can put the effort in to actually make this op knockout chem.

File: code/modules/reagents/chemistry/machinery/chem_dispenser.dm [hunk 7]
```diff
@@ -447,14 +481,20 @@
...
 		"lemonjuice",
 		"menthol"
-	)
+	) //prevents the soda machine from obtaining chemical upgrades. .
+	upgrade_reagents = null
+	upgrade_reagents2 = null
+	upgrade_reagents3 = null
 	emagged_reagents = list(
 		"thirteenloko",
+		"morphine",
```
Comment (silicons) [inline code/modules/reagents/chemistry/machinery/chem_dispenser.dm]: no thank you med already has this we don't need this as a morphine source

File: code/modules/reagents/chemistry/machinery/chem_dispenser.dm [hunk 8]
```diff
@@ -447,14 +481,20 @@
...
+		"uranium",
```
Comment (silicons) [inline code/modules/reagents/chemistry/machinery/chem_dispenser.dm]: no thank you infinite emps/uranium isn't a fun time

File: code/modules/reagents/chemistry/machinery/chem_dispenser.dm [hunk 9]
```diff
@@ -528,6 +575,7 @@
 	name = "mutagen dispenser"
 	desc = "Creates and dispenses mutagen."
 	dispensable_reagents = list("mutagen")
+	upgrade_reagents = null
```
Comment (silicons) [inline code/modules/reagents/chemistry/machinery/chem_dispenser.dm:577]: you forgot to null out upgrade_reagents 2/3

**Review by silicons — CHANGES_REQUESTED** [review]
why are emagged chems/upgrades in the same PR okay whatever
also updateparts or whatever needs to clear the upgraded chemicals if it downgrades a level instead of upgrades.

File: code/modules/reagents/chemistry/machinery/chem_dispenser.dm (hunk 9 above)
Comment (Trilbyspaceclone) [reply code/modules/reagents/chemistry/machinery/chem_dispenser.dm:577]: intented

**Review by silicons — APPROVED** [review]

**Pull request merged**